### PR TITLE
meson: add missing dependency to libdl

### PR DESCRIPTION
**Pull request by yuwata**
Follow-up for 1c20c9f4fce3b2eb501a776fb6025d6b5567fc00.

Fixes https://github.com/systemd/systemd/pull/31550#issuecomment-1980458377.

### CODE REVIEW

**Comment by DaanDeMeyer** [comment]
I'm a bit confused why all the CI jobs (except the Noble stuff) were green on that PR

File: src/basic/meson.build [hunk 1]
```diff
@@ -275,7 +275,8 @@ libbasic = static_library(
         basic_sources,
         fundamental_sources,
         include_directories : basic_includes,
-        dependencies : [libcap,
+        dependencies : [libdl,
+                        libcap,
```
Comment (YHNdnzj) [inline src/basic/meson.build]: Sort alphabetically?

**Review by YHNdnzj — APPROVED** [review]

**Comment by yuwata** [comment]
> I'm a bit confused why all the CI jobs (except the Noble stuff) were green on that PR.

Yeah, me too...

**Comment by teknoraver** [comment]
> I'm a bit confused why all the CI jobs (except the Noble stuff) were green on that PR.

And also I was compiling it fine under both x86_64 and arm64

**Pull request merged**